main.go

```diff
@@ -48,6 +49,7 @@ func main() {
 	var pull, noCache, push, debug bool
 	var buildNumber string
 	flag.StringVar(&buildNumber, constants.ArgNameBuildNumber, "0", fmt.Sprintf("Build number, this argument would set the reserved %s build environment.", constants.ExportsBuildNumber))
+	flag.StringVar(&dockerContextString, constants.ArgNameDockerContextString, "", "Working directory for the builder.")
```

Comment:
honestly before I review anything else, I would like to understand why we can't just have one parameter for all of this called `-c` - there should be no `gitURL`, `dockerfile`, `dockerContextString`, just one parameter. That's what I was expecting for this PR

Reply:
Those parameters can be removed. I just didn't include them here. If you want to control your own working directory where the files are checked out to you can still use them. 
However, this is a feature that we can decide to remove. Do we have to do them all in the same PR. I thought I was only trying to fix the scenario for runner. Which it currently only use a single parameter `archive`. We would replace it with `-context`

Reply:
I guess I don't understand the purpose of the PR, but in the description you put "[unify with Docker build]," this is a step backwards. Absolutely do not add more parameters, it's a complete mess already. Remove and rename them if you need to, this is what this PR reads like right now:

```
flag.StringVar(&dockerContextString, constants.ArgNameDockerContextString, "", "Working directory for the builder.")
flag.StringVar(&workingDir, constants.ArgNameWorkingDir, "", "Working directory for the builder.")
```